app.go

```diff
@@ -442,12 +448,20 @@ func (app *App) Stop(ctx context.Context) error {
 }
 
 // Done returns a channel of signals to block on after starting the
-// application. Applications listen for the SIGINT and SIGTERM signals; during
+// application, and adds that channel to the apps list of done channels.
+// Applications listen for the SIGINT and SIGTERM signals from the os, during
 // development, users can send the application SIGTERM by pressing Ctrl-C in
 // the same terminal as the running process.
+// Alternateively, a signal can be broadcast to all done channels manually by
+// using the Shutdown functionality (see the Shutdowner documentation for details)
+// todo(syrie): add Shutdowner documentation.
 func (app *App) Done() <-chan os.Signal {
 	c := make(chan os.Signal, 1)
 	signal.Notify(c, syscall.SIGINT, syscall.SIGTERM)
+
+	app.mu.Lock()
+	defer app.mu.Unlock()
```

Comment:
I agree that in this simpler case it makes sense to forgo the performance cost of defer since it doesn't add any readability benefit, but as a rule I've been used to defaulting to defer per the style guide. I'll put that through.